### PR TITLE
AYON: Tools enhancements

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Few enhancements and tweaks of AYON related tools.

## Additional info
- Fixed typo `set_name_filer` to `set_name_filter`.
- Renamed `get_current_project_name` to `get_selected_project_name` in ProjectsCombobox.
- Hierarchy model adds more information to `FolderItem` (path and folder type).
- Moved universal `TreeView` to tool utils from workfiles tool, and use it in FoldersWidget.
- `UNDERLINE_COLORS_ROLE` has bigger offset to avoid clashes with FoldersWidget roles.
- Projects/Folders/Tasks widgets have more public methods and signals.

Branch of this PR will be used as base for few of my future PRs.

## Testing notes:
All changes should not affect, any logic of tools. All tools loader, workfiles, launcher should work as expected.

### CODE REVIEW

File: openpype/tools/utils/views.py [hunk 1]
```diff
@@ -57,3 +59,68 @@ def paintEvent(self, event):
             self.paint_empty(event)
         else:
             super(TreeViewSpinner, self).paintEvent(event)
+
+
+class TreeView(QtWidgets.QTreeView):
```
Comment (BigRoy) [inline openpype/tools/utils/views.py:64]: Should this also get a better class name now? Seems overly generic for something that only adds the capability to be "flickable"?

It does add double click signals too though.

File: openpype/tools/ayon_utils/widgets/folders_widget.py [hunk 2]
```diff
@@ -323,16 +361,116 @@ def refresh(self):
...
+
+        Args:
+            enabled (bool): Enable deselectable mode.
+        """
+
+        self._folders_view.set_deselectable(enabled)
+
+    def _on_left_double_click(self):
+        self.double_clicked_left.emit()
+        self.double_clicked.emit()
```
Comment (BigRoy) [inline openpype/tools/ayon_utils/widgets/folders_widget.py:456]: Wouldn't this technically make more sense if it somehow would also emit along event data describing how the double click occurred? (e.g. middle-mouse?)

File: openpype/tools/ayon_utils/widgets/folders_widget.py [hunk 3]
```diff
@@ -306,7 +325,26 @@ def __init__(self, controller, parent, handle_expected_selection=False):
...
+        return self._folders_model.is_refreshing
+
+    @property
+    def has_content(self):
+        """Has at least one folder.
+
+        Returns:
+            bool: True if model has at least one folder.
+        """
+
```
Comment (BigRoy) [inline openpype/tools/ayon_utils/widgets/folders_widget.py:344]:
```suggestion
```
Just to match the style of method above.

**Comment by LiborBatek** [comment]
The ```Focus to context``` btn or how to call this function btn does not work in all cases....not keeping the proper ```task``` when hit and switch to the other ```task``` type in the ```asset``` instead, it works some times but some times not. Leading user to different ```task``` type which could lead to confusion and possible mistake when triggering ```Save As``` later on.
Focus_Btn_Issue

File: openpype/tools/ayon_utils/widgets/folders_widget.py [hunk 4]
```diff
@@ -306,7 +325,26 @@ def __init__(self, controller, parent, handle_expected_selection=False):
...
 
-    def set_name_filer(self, name):
+    @property
+    def is_refreshing(self):
+        """Model is refreshing.
+
+        Returns:
+            bool: True if model is refreshing.
+        """
+        return self._folders_model.is_refreshing
```
Comment (iLLiCiTiT) [inline openpype/tools/ayon_utils/widgets/folders_widget.py:335]:
```suggestion
        """

        return self._folders_model.is_refreshing
```

File: openpype/tools/utils/views.py (hunk 1 above)
Comment (iLLiCiTiT) [reply openpype/tools/utils/views.py:64]: And also contains logic of `DeselectableTreeView`.

File: openpype/tools/ayon_utils/widgets/folders_widget.py (hunk 2 above)
Comment (iLLiCiTiT) [reply openpype/tools/ayon_utils/widgets/folders_widget.py:457]: Will try to use `QtCore.Qt.Mouse...` constants...

Comment (BigRoy) [reply openpype/tools/ayon_utils/widgets/folders_widget.py:457]: Haha - couldn't we emit the original event data? :)

Comment (iLLiCiTiT) [reply openpype/tools/ayon_utils/widgets/folders_widget.py:457]: Yes, but I'm not sure if `Signal` can handle the constants (I mean in PySide6).

File: openpype/tools/utils/views.py [hunk 5]
```diff
@@ -101,11 +100,7 @@ def mousePressEvent(self, event):
         super(TreeView, self).mousePressEvent(event)
 
     def mouseDoubleClickEvent(self, event):
-        if event.button() == QtCore.Qt.LeftButton:
-            self.double_clicked_left.emit()
-
-        elif event.button() == QtCore.Qt.RightButton:
-            self.double_clicked_right.emit()
+        self.double_clicked.emit(event.button())
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/tools/utils/views.py:103]: I didn't thought about it. Could be changed...

**Review by LiborBatek — APPROVED** [review]
All seems fine.

**Pull request merged**